code/game/gamemodes/objective.dm

```diff
@@ -599,23 +605,33 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 	return FALSE
 
 /datum/objective/steal/proc/give_kit(obj/item/item_path)
-	var/I = new item_path
-	var/list/slots = list(
+	var/list/datum/mind/objective_owners = get_owners()
+	if(!length(objective_owners))
+		return
+
+	var/obj/item/item_to_give = new item_path
+	var/static/list/slots = list(
 		"backpack" = SLOT_HUD_IN_BACKPACK,
 		"left pocket" = SLOT_HUD_LEFT_STORE,
 		"right pocket" = SLOT_HUD_RIGHT_STORE,
 		"left hand" = SLOT_HUD_LEFT_HAND,
 		"right hand" = SLOT_HUD_RIGHT_HAND,
 	)
-	for(var/datum/mind/M in get_owners())
-		var/mob/living/carbon/human/H = M.current
-		var/where = H.equip_in_one_of_slots(I, slots)
+
+	for(var/datum/mind/kit_receiver_mind as anything in shuffle(objective_owners))
+		var/mob/living/carbon/human/kit_receiver = kit_receiver_mind.current
+		if(!kit_receiver)
+			continue
+
+		var/where = kit_receiver.equip_in_one_of_slots(item_to_give, slots)
 		if(where)
-			to_chat(H, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
-		else
-			to_chat(H, "<span class='userdanger'>Unfortunately, you weren't able to get a stealing kit. This is very bad and you should adminhelp immediately (press F1).</span>")
-			message_admins("[ADMIN_LOOKUPFLW(H)] Failed to spawn with their [item_path] theft kit.")
-			qdel(I)
+			to_chat(kit_receiver, "<br><br><span class='info'>In your [where] is a box containing <b>items and instructions</b> to help you with your steal objective.</span><br>")
```

Comment:
IMO, if there are multiple objective owners, we might want to tell other members of the objective that someone else on their team has the objective kit.